pageserver/src/restore_local_repo.rs

```diff
@@ -744,14 +741,14 @@ fn transaction_id_precedes(id1: TransactionId, id2: TransactionId) -> bool {
      * comparison.  If both are normal, do a modulo-2^32 comparison.
      */
 
-    if !(id1 >= pg_constants::FIRST_NORMAL_TRANSACTION_ID)
-        || !(id2 >= pg_constants::FIRST_NORMAL_TRANSACTION_ID)
+    if (id1 < pg_constants::FIRST_NORMAL_TRANSACTION_ID)
```

Comment:
The reason this was written this way was to match the code in Postgres TransactionIdPrecedes. But I agree it was confusing. I'm OK with changing this the way proposed here, but alternatively we could introduce a `TransactionIdIsNormal` function, like in Postgres, and use that here. Then this function would match the Postgres function more closely, and it would be easier to see that it's equivalent.

Reply:
Didn't know that, I like your idea about separate function, will do that. Will posgres_ffi be a suitable place for that?

Reply:
Yeah. Should probably move `transaction_id_precedes` there, too.

Reply:
done in #501